com.ibm.streamsx.topology/opt/python/include/splpy_funcop.h

```diff
@@ -114,6 +117,58 @@ class SplpyFuncOp : public SplpyOp {
                "streamsx.topology.runtime", wrapfn, appCallable, extraArg));
       }
 
+      /**
+       * Register a state handler for the operator.  The state handler
+       * handles checkpointing and supports consistent regions.  Checkpointing
+       * will be enabled only if the operator is stateful and if it can be
+       * saved and restored using dill.
+       */
+      void setupStateHandler() {
+        // If the value of the pyStateful param is true, create and register
+        // a state handler instance.  Otherwise, register a do-nothing
+        // state handler.
+        SPL::boolean stateful = static_cast<SPL::boolean>(op()->getParameterValues("pyStateful")[0]->getValue());
+        assert(!stateHandler);
+        // Ensure that callable() can be pickled before using it in a state
+        // handler.
+        PyObject * pickledCallable = NULL;
+        if (stateful) {
+          SplpyGIL lock;
+          PyObject * dumps = SplpyGeneral::loadFunction("dill", "dumps");
```

Comment:
Is it possible to use the existing generic function call function?

https://github.com/IBMStreams/streamsx.topology/blob/master/com.ibm.streamsx.topology/opt/python/include/splpy_general.h#L354

Reply:
The way that the existing callFunction method works is that on failure, it writes the python error information (including a traceback into dill code that is not likely to be useful to us or our users) into standard error, consumes the python error, and then throws an exception.  Using our application tracing, messages are written with timestamps, with the name of the operator in which they occurred, and the location in our source where they occurred.  The messages will also be shown in the correct sequence of the other trace messages.  I think without that information it will be difficult to diagnose the causes of python exceptions that may happen.